### ISSUE TITLE
Can not iterate object (but you can indeed)

### ISSUE THREAD

**Issue by garak** [kalessil/phpinspectionsea#803]
When I use `foreach` on an object, I'm getting an error saying
>Can not iterate myObject (must implement one of Iterator interfaces)

with a link to [documentation](https://github.com/kalessil/phpinspectionsea/blob/master/docs/types-compatibility.md#foreach-source-to-iterate-over)

This assumption is false, since PHP allows [iteration on any object](http://php.net/manual/en/language.oop5.iterations.php)

**Comment by rentalhost** [comment]
Confirmed with:

```php
/** @var \stdClass $myObject */
foreach ($myObject as $item) {
}
```

While PHP [**indeed allows that**](https://ideone.com/5hxqlj), I don't think that it should be allowed directly from this inspection - else, it will not make sense no more (it will be limited to inspect scalar types only). 

What PHP does here is iterate over all object properties, but it seems very confuses because it is scope dependent. If you iterate on a public scope, only public properties are iterated; if you iterate on a private scope, all properties are iterated and you can't identify what is public and what is private at this time.

My suggestion in this case is you defines your object like `@var mixed[] $myObject`, to you make sure that it is intentional for your code. Other option is use the [`ReflectionClass::getProperties()`](http://php.net/manual/en/reflectionclass.getproperties.php) that will offer a lot of features and will be scope independent.

So, my vote here is for an `wont-fix`.

What do you think @kalessil?

**Comment by garak** [comment]
While I agree that it could not be considered a best practice to iterate over an object, this can't be considered as an error, because it isn't.
So, at least the message should avoid "must" (and use "should" instead), and the severity should be lowered to a warning.

**Comment by kalessil** [comment]
Wording: `must` -> `should` needs update in any case =)
Error level: it can be changed in the inspection settings, so no need to change anything there.
Iterating object properties: will neglect the inspection intention, hence I'll update/create documentation where this promoting implicit object to array casting.

**Comment by kalessil** [comment]
@garak : your proposal turned to totally make sense =) Error level lowered to a warning, message updated and I also updated documentation. Thank you.
